lib/round.rb

```diff
@@ -0,0 +1,56 @@
+require './lib/turn'
+require './lib/deck'
+
+class Round
+  attr_reader :deck, :turns, :number_correct
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+    @number_correct = 0.0
+    @number_correct_by_category = {}
+    @total_turns_by_category = {}
+  end
+
+  def current_card
+    @deck.cards[0]
+  end
+
+  def take_turn(guess)
+    new_turn = Turn.new(guess, @deck.cards[0])
+
+    if !@total_turns_by_category.has_key?(@deck.cards[0].category)
+      @total_turns_by_category[@deck.cards[0].category] = 1.0
+      @number_correct_by_category[@deck.cards[0].category] = 0.0
+    else
+      @total_turns_by_category[@deck.cards[0].category] += 1
+    end
+
+    if new_turn.correct?
+      @number_correct += 1
+      @number_correct_by_category[@deck.cards[0].category] += 1
+    end
+
+    @turns << new_turn
+    @deck.cards.shift
+    new_turn.feedback
+  end
```

Comment:
There's a lot going on here, but your code does look good. You could bring lines 21-26 into a new helper method.